docs/csharp/codedoc.md

```diff
@@ -12,16 +12,464 @@ ms.devlang: dotnet
+        {
+
+        }
+    }
+
+<a name="param"></a>
+## &lt;param&gt;
+
+The &lt;param&gt; tag should be used in the comment for a method declaration to describe one of the parameters for the method.
+To document multiple parameters, use multiple &lt;param&gt; tags.
+
+    /// <summary>
+    /// This class does something
+    /// </summary>
+    public class SomeClass
+    {
+        /// <param name="a">The stuff to do</param>
+        public void DoStuff(int a)
+        {
+
+        }
+
+        /// <param name="a">A stuff to do</param>
+        /// <param name="b">Another stuff to do</param>
+        public void DoMoreStuff(int a, string b)
+        {
+
+        }
+    }
+
+The `name` parameter represents the name of a method parameter
+
+<a name="paramref"></a>
+## &lt;paramref&gt;
+
+The &lt;paramref&gt; tag gives you a way to indicate that a word in the code comments, 
+for example in a &lt;summary&gt; or &lt;remarks&gt; block refers to a parameter.
+
+    /// <summary>
+    /// This class does something
+    /// </summary>
+    public class SomeClass
+    {
+        /// <summary>
+        /// This method does stuff.
+        /// The <paramref name="a"/> takes a number
+        /// </summary>
+        /// <param name="a">The stuff to do</param>
+        public void DoStuff(int a)
+        {
+
+        }
+    }
+
+<a name="typeparam"></a>
+## &lt;typeparam&gt;
+
+The &lt;typeparam&gt; tag should be used in the comment for a generic type or method declaration to describe a type parameter. 
+Add a tag for each type parameter of the generic type or method.
+
+    /// <summary>
+    /// This class does something
+    /// </summary>
+    public class SomeClass
+    {
+        /// <summary>
+        /// Does generic stuff
+        /// </summary>
+        /// <typeparam name="T">The generic stuff to do</param>
+        public void DoGenericStuff<T>()
+        {
+
+        }
+    }
+
+<a name="typeparamref"></a>
+## &lt;typeparamref&gt;
+
+The &lt;typeparamref&gt; tag gives you a way to indicate that a word in the code comments, 
+for example in a &lt;summary&gt; or &lt;remarks&gt; block refers to a type parameter.
+
+    /// <summary>
+    /// This class does something
+    /// </summary>
+    public class SomeClass
+    {
+        /// <summary>
+        /// Does generic stuff <typeparamref name="T"/>
+        /// </summary>
+        /// <typeparam name="T">The generic stuff to do</param>
+        public void DoGenericStuff<T>()
+        {
+
+        }
+    }
+
+<a name="include"></a>
+## &lt;include&gt;
+
+The &lt;include&gt; tag lets you refer to comments in a separate XML file that describe the types and members 
+in your source code as opposed to placing documentation comments directly in your source code file.
+
+    
+    /// <include file='docs.xml' path='Docs/Members[@name="someclass"]/SomeClass/*' />
+    public class SomeClass
+    {
+        /// <include file='docs.xml' path='Docs/Members[@name="someclass"]/DoStuff/*' />
```

Comment:
Readers would gain a lot from an explanation of the "path" attribute here. It would make it much easier for them to use this in their own projects.